### PR TITLE
install fuse3 for 22.04, fuse for others

### PR DESCRIPTION
**Pull request by anujmaheshwari1**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh [hunk 1]
```diff
@@ -49,6 +49,10 @@ installDeps() {
     fi
 
     for apt_package in apt-transport-https ca-certificates ceph-common cgroup-lite cifs-utils conntrack cracklib-runtime ebtables ethtool fuse git glusterfs-client htop iftop init-system-helpers inotify-tools iotop iproute2 ipset iptables nftables jq libpam-pwquality libpwquality-tools mount nfs-common pigz socat sysfsutils sysstat traceroute util-linux xz-utils netcat dnsutils zip rng-tools kmod gcc make dkms initramfs-tools linux-headers-$(uname -r); do
+      if [[ "${apt_package}" == "fuse" && "${OSVERSION}" == "22.04" ]]; then
```
Comment (alexeldeib) [inline parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh:52]: we already special case ubuntu 22 for blobfuse above. why not reuse that?

this also adds a check for every single apt package if it's fuse and ubuntu which feels unnecessary...

Comment (anujmaheshwari1) [reply parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh]: discussed offline and addressed with refactor

File: parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh [hunk 2]
```diff
@@ -26,33 +26,34 @@ installDeps() {
...
     fi
 
-    for apt_package in apt-transport-https ca-certificates ceph-common cgroup-lite cifs-utils conntrack cracklib-runtime ebtables ethtool fuse git glusterfs-client htop iftop init-system-helpers inotify-tools iotop iproute2 ipset iptables nftables jq libpam-pwquality libpwquality-tools mount nfs-common pigz socat sysfsutils sysstat traceroute util-linux xz-utils netcat dnsutils zip rng-tools kmod gcc make dkms initramfs-tools linux-headers-$(uname -r); do
-      if ! apt_get_install 30 1 600 $apt_package; then
-        journalctl --no-pager -u $apt_package
-        exit $ERR_APT_INSTALL_TIMEOUT
-      fi
+    for apt_package in ${pkg_list[*]}; do
+        if ! apt_get_install 30 1 600 $apt_package; then
+            journalctl --no-pager -u $apt_package
```
Comment (alexeldeib) [inline parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh:54]: does this even do anything if you aren't installing a systemd service?

fine to leave as-is

**Review by alexeldeib — APPROVED** [review]
lgtm assuming green vhds ❤️

**Comment by anujmaheshwari1** [comment]
vhds all green, didnt trigger GPU ones because of quota. eyeballed both 1804 and 2204, has correct fuse/fuse3 and blobfuse2 in both.

**Pull request merged**